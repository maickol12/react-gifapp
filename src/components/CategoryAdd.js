import Reat, { useState } from 'react';
import PropTypes from 'prop-types';


const CategoryAdd = ({ setCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setCategory( value => [inputValue, ...value]);
            setInputValue('')
        }
    }
    return(
        <div>
            <form onSubmit={handleSubmit}>
                <p>{ inputValue }</p>
                <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange } />
                
            </form>
        </div>
    )
}

CategoryAdd.propTypes = {
    setCategory: PropTypes.func.isRequired
}

export default CategoryAdd;