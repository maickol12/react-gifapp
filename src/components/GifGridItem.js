import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {id,title,url} ) => { 
    return(
        <div className='card animate__fadeIn_10s'>
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.prototype = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}