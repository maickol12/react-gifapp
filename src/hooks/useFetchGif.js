import { useEffect, useState } from 'react';
import { getGifs } from '../helper/getGifs';

export const useFetchGifs = (category = 'Mamberroi') => {
    const [state, setState] = useState({
        data:[],
        loading:true
    }); 
    useEffect(() => {
        getGifs(category).then(
            imgs =>{ 
                setState({
                    data:imgs,
                    loading:false
                });
            }
        );
    },[category]);

    return state;
}