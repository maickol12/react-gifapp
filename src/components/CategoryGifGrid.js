//import React, { useState, useEffect } from 'react';
import React, { useState, useEffect } from 'react';
//import { getGifs } from '../helper/getGifs';
import { useFetchGifs } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const CategoryGifGrid = ({ category }) => {
    // const [images, setImages] = useState([]);
    // const [count, setCount ] = useState(0);
    
    const {data,loading} = useFetchGifs(category);

    

    // useEffect( () => {
    //     getGifs(category).then( setImages );
    // } , []);

   
    return(
        <div>
            <h3>Category { category }</h3>
            { loading && <p className="animate__animated animated__flash">Cargando</p>}
            <div className='card-grid'>
                {
                    data.map((img) => (
                        <GifGridItem 
                            key={img.id+img.title}
                            {...img} />
                    ))
                }
            </div>
        </div>
    )
};

CategoryGifGrid.propTypes = {
    category: PropTypes.string.isRequired
}