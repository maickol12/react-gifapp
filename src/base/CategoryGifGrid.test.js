import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { CategoryGifGrid } from '../components/CategoryGifGrid';
import { useFetchGifs } from '../hooks/useFetchGif';
jest.mock('../hooks/useFetchGif');

const category = 'Mamberroi';

describe('Probando en el <CategoryGifGrid />',() => {
    
    test('Debe de mostrarse correctamente',() => {
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wraper = shallow(<CategoryGifGrid category={ category } />);
        expect( wraper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs',() => {
        const gif = [{
            id: 1,
            title: 'https://hola.jpg',
            url: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gif,
            loading: false
        },
        {
            data: gif,
            loading: false
        });
        const wraper = shallow(<CategoryGifGrid category={ category } />);
        
        expect( wraper ).toMatchSnapshot();
        expect( wraper.find('p').exists()).toBe( false );
        expect( wraper.find('GifGridItem').length).toBe(gif.length);

    });
});