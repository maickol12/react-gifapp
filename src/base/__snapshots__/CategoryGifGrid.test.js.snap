// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando en el <CategoryGifGrid /> Debe de mostrarse correctamente 1`] = `
<div>
  <h3>
    Category 
    Mamberroi
  </h3>
  <p
    className="animate__animated animated__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</div>
`;

exports[`Probando en el <CategoryGifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<div>
  <h3>
    Category 
    Mamberroi
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id={1}
      key="1https://hola.jpg"
      title="https://hola.jpg"
      url="Cualquier cosa"
    />
  </div>
</div>
`;
